File: Java OOP-Feb2020/src/i_polymorphism_Lab/shapes/Rectangle.java
package i_polymorphism_Lab.shapes;

public class Rectangle extends Shape {
    private final Double height;
    private final Double width;

    public Rectangle(Double height, Double width) {
        this.height = height;
        this.width = width;
    }

    public Double getHeight() {
        return height;
    }

    public Double getWidth() {
        return width;
    }

    @Override
    protected Double calculatePerimeter() {
        return 2 * (height + width);
    }

    @Override
    protected Double calculateArea() {
        return height * width;
    }
}

File: Programming-for-QA-sep2023/20.Encapsulation-Inheritance-Abstraction-Polymorphism/05.Shapes/Rectangle.cs
using Shapes;
using System;
using System.Text;

public class Rectangle : IDrawable
{
    private int _width;
    private int _height;

    public Rectangle(int width, int height)
    {
        if (width <= 0 || height <= 0)
        {
            throw new ArgumentException();
        }
        this._width = width;
        this._height = height;
    }
    public void Draw()
    {
        StringBuilder sb = new StringBuilder();
        string firstLastLine = new string('*', _width);
        string midLine = "*";
        if (_width > 1)
        {
            midLine += new string(' ', _width - 2) + '*';
        }
        Console.WriteLine(firstLastLine);

        for (int i = 1; i <= _height - 2; i++)
        {
            Console.WriteLine(midLine);
        }

        Console.WriteLine(firstLastLine);
    }
}